tools/xtce_generation/telemetry_generator.py

```diff
@@ -26,298 +64,280 @@ class TelemetryGenerator:
+            )
+            parameter_type.attrib["name"] = name
+            parameter_type.attrib["signed"] = "true"
+            encoding = Et.SubElement(parameter_type, "xtce:IntegerDataEncoding")
+            encoding.attrib["sizeInBits"] = str(row["lengthInBits"])
+            encoding.attrib["encoding"] = "signed"
+
+        elif "BYTE" in row["dataType"]:
+            parameter_type = Et.SubElement(
+                self._parameter_type_set, "xtce:BinaryParameterType"
+            )
+            parameter_type.attrib["name"] = name
+
+            encoding = Et.SubElement(parameter_type, "xtce:BinaryDataEncoding")
+            encoding.attrib["bitOrder"] = "mostSignificantBitFirst"
 
-            parameter.attrib["parameterTypeRef"] = parameter_type_ref
+            size_in_bits = Et.SubElement(encoding, "xtce:SizeInBits")
+            fixed = True
+            if fixed:
+                fixed_value = Et.SubElement(size_in_bits, "xtce:FixedValue")
+                fixed_value.text = str(row["lengthInBits"])
+            else:
+                dynamic_value = Et.SubElement(size_in_bits, "xtce:DynamicValue")
+                param_ref = Et.SubElement(dynamic_value, "xtce:ParameterInstanceRef")
+                param_ref.attrib["parameterRef"] = "PKT_LEN"
+                linear_adjustment = Et.SubElement(
+                    dynamic_value, "xtce:LinearAdjustment"
+                )
+                linear_adjustment.attrib["slope"] = str(8)
+                # Intercept is the start bit of this parameter - (header bytes - 1) * 8
+                linear_adjustment.attrib["intercept"] = str(-(row["startBit"] - 40))
+
+        if row["convertAs"] == "ANALOG":
+            # Go look up the conversion in the AnalogConversions tab
+            # and add it to the encoding
+            self._add_analog_conversion(row, encoding)
+
+    def _add_analog_conversion(self, row, encoding):
+        # Look up the conversion in the AnalogConversions tab
+        analog_conversion = self.sheets["AnalogConversions"]
+        # conversion is a row from the AnalogConversions sheet
+        conversion = analog_conversion.loc[
+            (analog_conversion["mnemonic"] == row["mnemonic"])
+            & (analog_conversion["packetName"] == row["packetName"])
+        ].iloc[0]
+
+        # Create the Conversion element
+        default_calibrator = Et.SubElement(encoding, "xtce:DefaultCalibrator")
+        polynomial_calibrator = Et.SubElement(
+            default_calibrator, "xtce:PolynomialCalibrator"
+        )
+        # FIXME: Use lowValue / highValue from the conversion sheet
+        # FIXME: Handle segmented polynomials (only using first segment now)
+        for i in range(8):
+            col = f"c{i}"
+            if conversion[col] != 0:
+                term = Et.SubElement(polynomial_calibrator, "xtce:Term")
+                term.attrib["coefficient"] = str(conversion[col])
+                term.attrib["exponent"] = str(i)
```

Comment:
I don't enough about how xml does calculation but it seems much nicer if we can do it here like you have